### ISSUE TITLE
Converters still missing textures

### ISSUE THREAD

**Issue by othermusketeer** [malte0811/IndustrialWires#5]
While they appear with textures in my inventory, they show up with the "missing texture" texture when placed in the world. Happens with and without optifine, with and without resource packs. After fiddling with the converter parts for 30min, I successfully placed everything in the right order/orientation and can confirm that they do function. As there were no crashes, Im not sure if you want any logs or images from me.

The mod seems quite nice. Thank you for making it. I hope the textures issue is eventually worked out so I can use it easily. I wish I knew minecraft modding (a little bit of Kerbal Space Program and Dwarf Fortress, but not minecraft) to be able to help. Let me know if you need anything from me.

Minecraft 1.10.2, Forge
Forge 12.18.3.2239
IndustrialWires-1.2-6
industrialcraft-2-2.6.142-ex110
InventoryTweaks-1.61-58
OptiFine_1.10.2_HD_U_D6
Build: 20170112-180947
OS: Windows 7 (amd64) version 6.1
Java: 1.8.0_91, Oracle Corporation
OpenGL: GeForce GTX 260/PCIe/SSE2, version 3.3.0, NVIDIA Corporation
OpenGL Version: 3.3
Maximum texture size: 8192x8192
(other mods present. Modified Infinity Lite install)

**Comment by malte0811** [comment]
This seems to be caused by Chisel, no idea how though.
@tterrag1098 any suggestions? I feel like I am missing something stupid here. [Blockstate JSON](https://github.com/malte0811/IndustrialWires/blob/master/src/main/resources/assets/industrialwires/blockstates/mechanical_converter.json), [Model JSON](https://github.com/malte0811/IndustrialWires/blob/master/src/main/resources/assets/industrialwires/models/block/iwSixSides.json). IE has some blocks using practically the same JSON's, those seem to be fine...

**Comment by tterrag1098** [comment]
Can't tell you much without an error log

**Comment by malte0811** [comment]
There doesn't seem to be any error log, I looked for "industrialwires" and "motor" (one of the broken blocks) in the client fml log and didn't find anything except for some messages saying that the item models were being redirected to the block state json, which they should be. The textures don't show up as missing in the missing textures output when first entering the main menu.

**Comment by tterrag1098** [comment]
The way your converter block uses extended states is strange. You shouldn't be modifying the base blockstate in that method. I don't even see the purpose of your getExtendedState code at all. I would try first getting rid of that to see if it solves the problem

**Comment by malte0811** [comment]
That fixed it, thanks. I don't know why I put that stuff in `getExtendedState`, I probably wasn't paying too much attention to it when I did so. I will push a fix and publish a new release in a bit.
The reason why removing the `getExtendedState` method fixed it is not that I was modifying the non-extended block state in `getExtendedBlockState` since the same modification was done in `getActualBlockState`: It was the fact that I made `getActualState` return `getExtendedState` which calls the superclass method. Normally that method just returns the `state` parameter, but with Chisel it wraps the parameter in some Chisel specific extended block state. And since the block states used to map states to models don't equal (`hashCode`/`equals`) their wrapped states MC was unable to retrieve a model for that state.
